CONTRIBUTING.md

```diff
@@ -12,33 +12,45 @@ We welcome contributions for bug fixes, new efficient methods you'd like to cont
 
 Have a new algorithm you'd like to contribute to the library as part of your research? We welcome any PRs, and recommend filing an issue with the proposed method or reaching out on Slack first!
 
+## Prerequisites
+
+To set up the development environment in your local box, run the command below. 
+
+1\. This should bring most the dependencies needed for testing and linting the code:
+
+```bash
+pip install -e .[dev]
+```
+
+2\. Next, you need to install `pyright` (we have not included it in `setup.py` because we have found that pyright is unreliable when it comes to version pinning). For this, you will need to first install NodeJS, more instructions can be found at https://nodejs.org/en/download/package-manager/. Then, to install pyright run:
+
+```bash
+# NOTE. You might need sudo permissions.
+npm install -g pyright@1.1.224
+```
 
 ## Submitting a contribution
 
 To submit a contribution:
 
-1. Fork a copy of the [Composer](https://github.com/mosaicml/composer) library to your own account.
+1\. Fork a copy of the [Composer](https://github.com/mosaicml/composer) library to your own account.
```

Comment:
Adds manual numbering here so that code block doesn't break the document numbering order.